mutex_deadlocks.go

```diff
@@ -0,0 +1,25 @@
+//go:build deadlocks || race
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package cni
+
+import (
+	"github.com/sasha-s/go-deadlock"
+)
+
+type RWMutex = deadlock.RWMutex
```

Comment:
Using an interface for `sync.RWMutex` seems easier to manage and test explicitly for than using build tags?

Reply:
Nit preference?

Reply:
Well my point is you can have an interface in the struct and set it to use the deadlock.RWMutex in a test.

Reply:
it's possible, but build tags allows to _exclude_ deadlock.RWMutex completely from the build if it's not race-enabled or has an explicit `deadlocks` build tag. 

plus interface indirection has a slight performance hit

this change is equivalent to what it was before if not using race/build tag